### ISSUE TITLE
SSH Zugriff über Passwort deaktivieren (SSH-Key only)

### ISSUE THREAD

**Issue by DaRe92** [mittwald/feature-requests#41]
**Welches Problem möchtest du lösen? Wann tritt es auf?**
Aktuell kann man sich per Passwort auf den Servern anmelden. 

**Welche Lösungsideen hast du?**
Zugriff per Passwort generell verbieten und nur per SSH-Key erlauben. Alternativ auf Organisationsebene in den Einstellungen eine Möglichkeit den Zugriff per Passwort zu verbieten.

**Comment by paul-hph** [comment]
👍 
> Alternativ auf Organisationsebene in den Einstellungen eine Möglichkeit den Zugriff per Passwort zu verbieten.

**Comment by vogelor** [comment]
Ich denke, das ist schwierig.
Ich verstehe das Problem - und finde Keys auch gut.
Aber ich denke nicht, dass ich das an meine Kunden (oder an Web-Agenturen, die nur Designer haben) hin bekomme.
Die versehen gar nicht, was ich von denen will...

**Comment by DaRe92** [comment]
> Ich denke, das ist schwierig.
> Ich verstehe das Problem - und finde Keys auch gut.
> Aber ich denke nicht, dass ich das an meine Kunden (oder an Web-Agenturen, die nur Designer haben) hin bekomme.
> Die versehen gar nicht, was ich von denen will...

Jemand der mit SSH-Keys nichts anfangen kann hat meiner Meinung nach auch nichts per SSH auf dem Server zu suchen! Davon abgesehen könnte man es ja auch auf Organisationsebene einstellbar machen.

**Comment by vogelor** [comment]
@DaRe92 ich stimme dir zu 100% zu. Aber ich kann schlecht sagen, wenn du nicht weißt, was ein SSH-Key ist bekommst du von mir keinen Zugang zu deinem Server. Dann ist der Kunde schneller weg, als ich "Hallo" sagen kann. Das ist das Problem...

**Comment by patrickhilker** [comment]
Um das kurz etwas zu konkretisieren: Es geht hier um den **mStudio-User, der sich per SSH anmeldet**, oder? Denn die zusätzlichen SSH-User, die bspw. für Deployments o.ä. angelegt werden können, unterstützen es bereits, dass sie kein Passwort haben (s. Screenshot unten). Wurde der User initial mit Passwort angelegt, kann das Passwort auch nachträglich entfernt werden.

Oder geht es eher um eine Einstellung am Projekt?

<img width="954" alt="image" src="https://github.com/mittwald/feature-requests/assets/5042280/141d9fd6-bdc8-463d-a86a-20d6a620cb74">

**Comment by mpsn** [comment]
Am Besten wäre es, wenn man das pro Projekt konfigurieren könnte, wahrscheinlich ist es aber schwer umsetzbar. Konfiguration auf Server-Ebene würde ich auch gut finden.
